src/containers/Review/assignment/Assignment.tsx

```diff
@@ -1,162 +1,48 @@
-    Object.values(fullApplicationStructure.sections).reduce(
-      (assignedSections, { details: { code } }) => ({
-        ...assignedSections,
-        [code]: { newAssignee: undefined },
-      }),
-      {}
-    )
-  )
-
-  const submitAssignments = () => {
-    // Re-assignment - grouping sections that belong to same (new) assignment
-    let reassignmentGroupedSections: {
-      sectionCodes: string[]
-      unassignmentId: number
-      reassignment: AssignmentDetails
-    }[] = []
 
-    const reassignedSections = Object.entries(assignedSections).filter(
-      ([_, { previousAssignee }]) => !!previousAssignee
-    )
+  const shouldUpdate = !Object.keys(structures).some((key) => Number(key) === assignment.id)
 
-    reassignedSections.forEach(([sectionCode, sectionAssignee]) => {
-      const { newAssignee, previousAssignee } = sectionAssignee
-      const found = reassignmentGroupedSections.find(
-        ({
-          reassignment: {
-            reviewer: { id },
-          },
-        }) => id === newAssignee
-      )
-      const unassignment = assignmentsByStageAndLevel.find(
-        ({ reviewer: { id } }) => id === previousAssignee
-      )
+  const { fullReviewStructure, error } = useGetReviewStructureForSections({
+    reviewAssignment: assignment,
+    fullReviewStructure: structure,
+    awaitMode: !shouldUpdate,
+  })
 
-      if (found) {
-        found.sectionCodes.push(sectionCode)
-        if (!!unassignment && found?.unassignmentId != unassignment.id)
-          console.log('Unassignment for more than one previous assignee - Failed')
-      } else {
-        const reassignment = assignmentsByStageAndLevel.find(
-          ({ reviewer: { id } }) => id === newAssignee
-        )
-        if (reassignment && unassignment)
-          reassignmentGroupedSections.push({
-            sectionCodes: [sectionCode],
-            reassignment,
-            unassignmentId: unassignment.id,
-          })
-      }
-    })
-    reassignmentGroupedSections.forEach(({ sectionCodes, reassignment, unassignmentId }) =>
-      reassignSections({ sectionCodes, reassignment, unassignmentId })
-    )
+  if (error) return <Message error title={strings.ERROR_GENERIC} list={[error]} />
 
-    // First assignment - grouping sections that belong to same assignment
-    let assignmentGroupedSections: {
-      sectionCodes: string[]
-      assignment: AssignmentDetails
-    }[] = []
+  if (fullReviewStructure && shouldUpdate) {
+    // Token required because this was running on loop trying to set the Assignments many times
+    if (shouldUpdate && !syncToken) {
+      setReviewStructures({ type: 'getSyncToken' })
 
-    const firstAssignedSections = Object.entries(assignedSections).filter(
-      ([_, { previousAssignee }]) => previousAssignee === undefined
-    )
+      setReviewStructures({
+        type: 'addReviewStructure',
+        reviewStructure: fullReviewStructure,
+        assignment,
+      })
```

Comment:
Oooh, I think I've figured it out -- `useEffect` to the rescue!

Don't need the tokens at all.

The whole "Assignment" component can just be:
```ts
// Component only used to update the contxt with reviewStructuresState (run one instance of hook)
const Assignment: React.FC<ReviewHomeProps> = ({ assignment, structure }) => {
  const {
    reviewStructuresState: { syncToken, structures },
    setReviewStructures,
  } = useReviewStructureState()
  const { strings } = useLanguageProvider()

  const shouldUpdate = !Object.keys(structures).some((key) => Number(key) === assignment.id)

  const { fullReviewStructure, error } = useGetReviewStructureForSections({
    reviewAssignment: assignment,
    fullReviewStructure: structure,
    awaitMode: !shouldUpdate,
  })

  useEffect(() => {
    if (fullReviewStructure && shouldUpdate) {
      setReviewStructures({
        type: 'addReviewStructure',
        reviewStructure: fullReviewStructure,
        assignment,
      })
    }
  }, [fullReviewStructure])

  if (error) return <Message error title={strings.ERROR_GENERIC} list={[error]} />

  return null
}
```

Loads properly, seems to work and no React error! (there's still a missing key error, but I've mentioned that elsewhere)

Reply:
AMAZING!!! I'll add this right away and test. I guess after that we can probably start workiing on merging all this together? I still would like to add a solution for issue #1109 and #1110, but I would like more if this can be done in the next version, based from `develop`.